doc(dev): add plugins sequence diagram (#5523)
centreon/centreon-plugins: doc/en/developer/sequence_diagram.md

`mode`, `perfdata`, `values` and  `output`) handle them.

```mermaid
sequenceDiagram
    counter.pm ->> +themode.pm: $self->set_counters(%options);
        Note left of themode.pm: set_counters() must<br/>define the counter's<br/>structure
    themode.pm -->> -counter.pm: Back from set_counters()
```
    
### manage_selection()

This method is the one that will be the most specific to the plugin and the mode you're working on since it will have to
cope with both the protocol and the data structures in input and in output.

```mermaid
sequenceDiagram
    counter.pm ->> +themode.pm: $self->manage_selection(%options)
        Note left of themode.pm: Gathers the data and stores<br/>them in the counters<br/>structure
    themode.pm -->> -counter.pm: Back from manage_selection()
```

### Callback functions

> The callback functions must be defined in _themode.pm_ but actually apply to objects of **values** class.

#### closure_custom_calc

```mermaid
sequenceDiagram
    values.pm ->> themode.pm: $self->{closure_custom_calc}->(...)
```

#### closure_custom_output

```perl

> return $self->{closure_custom_output}->($self);

Author: WTF

        counter.pm -->> -themode.pm: Back to check_options()
        Note right of themode.pm: Checks that the mode-specific<br/>options are valid
    themode.pm -->> -script_custom.pm: Back to init() from check_options()
```

### set_counters()

This method will describe how you decide to structure the collected data in order to let the plugins classes (`counter`, 
`mode`, `perfdata`, `values` and  `output`) handle them.

```mermaid
sequenceDiagram
    counter.pm ->> +themode.pm: $self->set_counters(%options);
        Note left of themode.pm: set_counters() must<br/>define the counter's<br/>structure
    themode.pm -->> -counter.pm: Back from set_counters()
```
    
### manage_selection()

This method is the one that will be the most specific to the plugin and the mode you're working on since it will have to
cope with both the protocol and the data structures in input and in output.

```mermaid
sequenceDiagram
    counter.pm ->> +themode.pm: $self->manage_selection(%options)
        Note left of themode.pm: Gathers the data and stores<br/>them in the counters<br/>structure
    themode.pm -->> -counter.pm: Back from manage_selection()
```

### Callback functions

> > The callback functions must be defined in _themode.pm_ but actually apply to objects of **values** class.

Author: ```suggestion
> The callback functions must be defined in the corresponding _themode.pm_ file but they actually apply to objects with the **values** class (defined in the **values.pm** file). You should bear it in mind while writing these functions to know what data you can access.
```

#### closure_custom_calc

This function receives the raw values in a hash reference `$options{new_datas}` and must feed `$self->{result_values}`.

> The default is `centreon::plugins::templates::catalog_functions::catalog_status_calc()`.

Author: ```suggestion
The default function that will be called is `centreon::plugins::templates::catalog_functions::catalog_status_calc()`.
```

```mermaid
sequenceDiagram
    values.pm ->> themode.pm: $self->{closure_custom_calc}->(...)
```

#### closure_custom_output

> Must be written as a method of `values` package/class.

Author: ```suggestion
```

> Using data stored as attributes of `$self->{result_values}` where the available keys are the strings listed in the
> `key_values` entries, the method must return a string.

Author: ```suggestion
This method must return a string to be displayed in the output using data stored as attributes of `$self->{result_values}` where the available keys are the strings listed in the `key_values` entries.
```
